### PR TITLE
Resupply & Integrations Update

### PR DESCRIPTION
**Pull request by EidolonForgery**
Adds SWS custom supply pods and removes issues with my own lack of rebasing :)

### CODE REVIEW

File: addons/resupply/config.cpp [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+            ITEM_XX(ACE_bloodIV,30);
+            ITEM_XX(ACE_elasticBandage,100);
+            ITEM_XX(ACE_quikclot,40);
+        };
+        class TransportWeapons {
+            WEAP_XX(SWS_Weapon_M24RL_D_Loaded,8);
+        };
+    };
+    class OPTRE_Ammo_SupplyPod_Empty;
+    class SWS_Ammo_SupplyPod_Empty : OPTRE_Ammo_SupplyPod_Empty {
```
Comment (WitchFreya) [inline addons/resupply/config.cpp:86]: Do you want an empty supply pod to be an option? If not, make this ITEM_META(0); and then use SCOPE(2); in the spots that inherit from it.

If it complains about something with scope = 0 try ITEM_META(1); instead. I've had weird issues with it.

File: addons/resupply/config.cpp [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+		hiddenselectionstextures[] = {QPATHTOF(data\sws_supplypod_base_co.paa)};
+
+        OBJ_CATEGORY(SWSResupply);
+        class TransportMagazines{};
+        class TransportWeapons{};
+        class TransportItems{};
+    };
+    class SWS_Ammo_SupplyPod_Basic : SWS_Ammo_SupplyPod_Empty {
+        displayName = "[SWS] Basic Resupply";
+        author = "Black";
```
Comment (WitchFreya) [inline addons/resupply/config.cpp:120]: Remove this line; it's inherited.

File: addons/resupply/config.cpp [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+                MAG_XX(SWS_Magazine_60Rnd_65x39_Caseless_Tracers_Blue,5);
+                MAG_XX(OPTRE_36Rnd_95x40_Mag,8);
+                MAG_XX(OPTRE_60Rnd_5x23mm_Mag,10);
+                MAG_XX(OPTRE_15Rnd_762x51_Mag,8);
+            };
+    };
+    class SWS_Ammo_SupplyPod_Medical : SWS_Ammo_SupplyPod_Empty {
+        displayName = "[SWS] Medical Resupply";
+        author = "Black";
+        hiddenselectionstextures[] = {QPATHTOF(data\sws_supplypod_medical_co.paa)};
```
Comment (WitchFreya) [inline addons/resupply/config.cpp:137]:
```suggestion
        hiddenSelectionsTextures[] = {QPATHTOF(data\sws_supplypod_medical_co.paa)};
```

File: addons/resupply/config.cpp [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        class TransportItems {
+            ITEM_XX(ACE_bloodIV,10);
+            ITEM_XX(ACE_elasticBandage,40);
+            ITEM_XX(ACE_quikclot,10);
+        };
+    };
+    class SWS_Ammo_SupplyPod_Howdy : SWS_Ammo_SupplyPod_Empty {
+        displayName = "[SWS] Howdy Resupply";
+        author = "Black";
+        hiddenselectionstextures[] = {QPATHTOF(data\sws_supplypod_base_co.paa)};
```
Comment (WitchFreya) [inline addons/resupply/config.cpp:147]:
```suggestion
        hiddenSelectionsTextures[] = {QPATHTOF(data\sws_supplypod_base_co.paa)};
```

File: addons/resupply/config.cpp [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+        {
+            WEAP_XX(SWS_Weapon_M24RL_D_Loaded,4);
+        };
+    };
+    class SWS_Ammo_SupplyPod_Spnkr : SWS_Ammo_SupplyPod_Empty {
+        displayName = "[SWS] SPNKR Resupply";
+        author = "Black";
+        hiddenselectionstextures[] = {QPATHTOF(data\sws_supplypod_spnkr_co.paa)};
+        class TransportWeapons
+            {
```
Comment (WitchFreya) [inline addons/resupply/config.cpp:158]:
```suggestion
        {
```

File: addons/resupply/config.cpp [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+        author = "Black";
+        hiddenselectionstextures[] = {QPATHTOF(data\sws_supplypod_base_co.paa)};
+        class TransportWeapons
+        {
+            WEAP_XX(SWS_Weapon_M24RL_D_Loaded,4);
+        };
+    };
+    class SWS_Ammo_SupplyPod_Spnkr : SWS_Ammo_SupplyPod_Empty {
+        displayName = "[SWS] SPNKR Resupply";
+        author = "Black";
```
Comment (WitchFreya) [inline addons/resupply/config.cpp:155]:
```suggestion
```

File: addons/resupply/config.cpp [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+                MAG_XX(OPTRE_M41_Twin_HEAP,2);
+                MAG_XX(OPTRE_M41_Twin_HEAT,2);
+            };
+    };
+    class SWS_Ammo_SupplyPod_SM247 : SWS_Ammo_SupplyPod_Empty {
+        displayName = "[SWS] SM247 Resupply";
+        author = "Black";
+        hiddenselectionstextures[] = {QPATHTOF(data\sws_supplypod_mg_co.paa)};
+        class TransportMagazines
+            {
```
Comment (WitchFreya) [inline addons/resupply/config.cpp:176]:
```suggestion
        {
```

File: addons/resupply/config.cpp [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+                MAG_XX(SWS_Magazine_200Rnd_762x51_Mag,4);
+                MAG_XX(SWS_Magazine_200Rnd_762x51_Mag_Tracer,4);
+                MAG_XX(SWS_Magazine_400Rnd_762x51_Mag,2);
+                MAG_XX(SWS_Magazine_400Rnd_762x51_Mag_Tracer,2);
+            };
```
Comment (WitchFreya) [inline addons/resupply/config.cpp:181]:
```suggestion
        };
```

File: addons/resupply/config.cpp [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+                MAG_XX(OPTRE_M41_Twin_HEAT_SACLOS,4);
+                MAG_XX(OPTRE_M41_Twin_HEAT_SALH,2);
+                MAG_XX(OPTRE_M41_Twin_HEAT_Guided,2);
+                MAG_XX(OPTRE_M41_Twin_HEAP,2);
+                MAG_XX(OPTRE_M41_Twin_HEAT,2);
+            };
+    };
+    class SWS_Ammo_SupplyPod_SM247 : SWS_Ammo_SupplyPod_Empty {
+        displayName = "[SWS] SM247 Resupply";
+        author = "Black";
```
Comment (WitchFreya) [inline addons/resupply/config.cpp:173]:
```suggestion
```

File: addons/resupply/config.cpp [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+                MAG_XX(OPTRE_M41_Twin_HEAT_Thermal_ProximityFuse,2);
+                MAG_XX(OPTRE_M41_Twin_HEAT_SACLOS,4);
+                MAG_XX(OPTRE_M41_Twin_HEAT_SALH,2);
+                MAG_XX(OPTRE_M41_Twin_HEAT_Guided,2);
+                MAG_XX(OPTRE_M41_Twin_HEAP,2);
+                MAG_XX(OPTRE_M41_Twin_HEAT,2);
+            };
+    };
+    class SWS_Ammo_SupplyPod_SM247 : SWS_Ammo_SupplyPod_Empty {
+        displayName = "[SWS] SM247 Resupply";
```
Comment (WitchFreya) [inline addons/resupply/config.cpp:172]:
```suggestion
        displayName = QNAME(SM247 Resupply);
```

File: addons/resupply/config.cpp [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+        displayName = "[SWS] Howdy Resupply";
+        author = "Black";
+        hiddenselectionstextures[] = {QPATHTOF(data\sws_supplypod_base_co.paa)};
+        class TransportWeapons
+        {
+            WEAP_XX(SWS_Weapon_M24RL_D_Loaded,4);
+        };
+    };
+    class SWS_Ammo_SupplyPod_Spnkr : SWS_Ammo_SupplyPod_Empty {
+        displayName = "[SWS] SPNKR Resupply";
```
Comment (WitchFreya) [inline addons/resupply/config.cpp:154]:
```suggestion
        displayName = QNAME(SPNKR Resupply);
```

File: addons/resupply/config.cpp [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+        author = "Black";
+        hiddenselectionstextures[] = {QPATHTOF(data\sws_supplypod_medical_co.paa)};
+        class TransportItems {
+            ITEM_XX(ACE_bloodIV,10);
+            ITEM_XX(ACE_elasticBandage,40);
+            ITEM_XX(ACE_quikclot,10);
+        };
+    };
+    class SWS_Ammo_SupplyPod_Howdy : SWS_Ammo_SupplyPod_Empty {
+        displayName = "[SWS] Howdy Resupply";
```
Comment (WitchFreya) [inline addons/resupply/config.cpp:145]:
```suggestion
        displayName = QNAME(Howdy Resupply);
```

File: addons/resupply/config.cpp [hunk 13]
```diff
@@ -0,0 +1,183 @@
...
+		hiddenselections[] = {"camo"};
+		hiddenselectionstextures[] = {QPATHTOF(data\sws_supplypod_base_co.paa)};
+
+        OBJ_CATEGORY(SWSResupply);
+        class TransportMagazines{};
+        class TransportWeapons{};
+        class TransportItems{};
+    };
+    class SWS_Ammo_SupplyPod_Basic : SWS_Ammo_SupplyPod_Empty {
+        displayName = "[SWS] Basic Resupply";
```
Comment (WitchFreya) [inline addons/resupply/config.cpp:119]:
```suggestion
        displayName = QNAME(Basic Resupply);
```

File: addons/resupply/config.cpp [hunk 14]
```diff
@@ -0,0 +1,183 @@
...
+            ITEM_XX(ACE_elasticBandage,100);
+            ITEM_XX(ACE_quikclot,40);
+        };
+        class TransportWeapons {
+            WEAP_XX(SWS_Weapon_M24RL_D_Loaded,8);
+        };
+    };
+    class OPTRE_Ammo_SupplyPod_Empty;
+    class SWS_Ammo_SupplyPod_Empty : OPTRE_Ammo_SupplyPod_Empty {
+        displayName = "[SWS] Supply Pod (Empty)";
```
Comment (WitchFreya) [inline addons/resupply/config.cpp:87]:
```suggestion
        displayName = QNAME(Supply Pod (Empty));
```

File: addons/resupply/config.cpp [hunk 15]
```diff
@@ -0,0 +1,183 @@
...
+        displayName = "Supply Pods & Boxes"; 
+    };
+};
+
+class CfgVehicles {
+    
+    class NATO_Box_Base;
+	class Box_SWS_TeamBox : NATO_Box_Base {
+        displayName = "[SWS] Team Supply Box";
+        dlc = "SWS";
```
[the remark this answers is not in the document]
Comment (WitchFreya) [reply addons/resupply/config.cpp:37]: This means removing the following properties:
- author
- scope
- scopeCurator
- dlc

File: addons/resupply/config.cpp [hunk 16]
```diff
@@ -0,0 +1,183 @@
...
+            "ace_medical_treatment"};
+        author = QUOTE(AUTHOR);
+        VERSION_CONFIG;
+    };
+};
+
+class CfgEditorCategories {
+	class SWSObjects
+	{
+		displayName = "[SWS] Objects";
```
Comment (WitchFreya) [inline addons/resupply/config.cpp:23]:
```suggestion
		displayName = QNAME(Objects); 
```

File: addons/resupply/config.cpp [hunk 17]
```diff
@@ -0,0 +1,183 @@
...
+{
+    class ADDON
+    {
+        name = QUOTE(COMPONENT);
+        units[] = {"Box_SWS_TeamBox","SWS_Ammo_SupplyPod_Empty","SWS_Ammo_SupplyPod_Basic","SWS_Ammo_SupplyPod_Medical","SWS_Ammo_SupplyPod_Howdy"};
+        weapons[] = {};
+        requiredVersion = REQUIRED_VERSION;
+        requiredAddons[] = {
+            "OPTRE_Misc",
+            "SWS_Weapons",
```
Comment (WitchFreya) [inline addons/resupply/config.cpp:13]:
```suggestion
            "sws_weapons",
```
Won't break anything, but the addon names are lowercased.

**Review by WitchFreya — COMMENTED** [review]
Some changes.

Another thing to consider that isn't entirely necessary; remove the `sws_` prefix from all the paa files. Up to you.

**Review by WitchFreya — CHANGES_REQUESTED** [review]
See comments; accidentally hit the 'comment' button instead of 'Request changes'.

File: addons/resupply/config.cpp (hunk 15 above)
[the remark this answers is not in the document]
Comment (EidolonForgery) [reply addons/resupply/config.cpp:37]: Attempting to use the SCOPE macro made the objects inaccessible last time but i can't remember if that was due to some archaic bullshit with the SIDE property. Will investigate.

File: addons/resupply/config.cpp (hunk 1 above)
Comment (EidolonForgery) [reply addons/resupply/config.cpp:86]: An empty supply pod is desired for architect reasons; Gotta have that SWS branding.

Comment (WitchFreya) [reply addons/resupply/config.cpp:86]: Gotcha, that's fine then.

**Comment by EidolonForgery** [comment]
All should be compliant: Please verify correct usage of CLASSES_VEHICLES, however. I'm not good at computer.

**Comment by EidolonForgery** [comment]
Hit the wrong button. Pls to be merging if it looks ok.

**Review by WitchFreya — CHANGES_REQUESTED** [review]
You'll wanna take a look at how CLASSES_VEHICLES and the Cfg files are laid out on the other addons; they're defined in the CfgVehicles.inc file and then included at the top of the config.cpp so they can be used.

**Review by WitchFreya — APPROVED** [review]

**Pull request merged**